src/main/java/algorithms/curated170/medium/CampusBikes2.java

```diff
@@ -0,0 +1,89 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CampusBikes2 {
+
+      private int dist(int wPos, int bPos) {
+            return Math.abs(workers[wPos][0] - bikes[bPos][0]) + Math.abs(workers[wPos][1] - bikes[bPos][1]);
+      }
+
+      private int targetBitmask;
+      int[][] workers, bikes;
+      int[][] dp;
+
+      public int assignBikes(int[][] workers, int[][] bikes) {
+            dp = new int[workers.length][1<<bikes.length];
```

Comment:
It looks we may not really need which specific worker has the bike. If we use "mask" as a database to persist the state of bike assignment, dp can be 1D.

Reply:
True, this holds valid because only n many digits could have been 0'ed when 2 workers are assigned;

Reply:
Added.